İŞ PROBLEMİ B 

Bir e-ticaret şirketi, müşterilerine yönelik iki farklı e-posta pazarlama kampanyasını test etmek istemektedir. Bunları "Kampanya A" ve "Kampanya B" olarak adlandırabiliriz. Kampanya A, ürünlerin tanıtıldığı geleneksel bir e-posta içeriği sunarken, Kampanya B interaktif ve kişiselleştirilmiş içerikler içermektedir. Bu deney kapsamında müşteriler rastgele olarak kampanyalardan birine atanmış ve her birine e-posta gönderilmiştir. Bu işlemler sonucunda üretilen veriler şirketin bilgi işlem birimi tarafından email_marketing.csv isimli bir dosya ile size sunulmaktadır. 

Şirket, hangi kampanyanın satın alma miktarlarını artırdığını belirlemek istiyor. Bu kapsamda tarafınızdan yapılması beklenenler aşağıda belirtilmektedir.

In [ ]:
SORU 1 - Yalnızca gerekli kütüphaneleri ve testleri çalışma ortamına yükleyiniz. (5 puan) - Kullanılmadığı halde yüklenmiş kütüphane ve testlerin her biri için 1 puanınız düşürülecektir

In [3]:
import pandas as pd
from scipy.stats import shapiro, levene, ttest_ind,  mannwhitneyu

SORU 2 - Veri setini çalışma ortamına yükleyiniz. (5 puan)

In [2]:
df = pd.read_csv('email_marketing.csv')

SORU 3a - Veri setinin baştan ve sondan ilk 10 satırını görüntüleyiniz. (5 puan)

In [ ]:
df.head(10)
df.tail(10)

In [ ]:
SORU 3b - Karşılaştırma yaptığınız gruplar içerisinde, önemli gördüğünüz hususlar (örneğin, grupların ortalama değerleri gibi) hakkında yorum yapınız. (10 puan)

In [5]:
df.groupby('Kampanya').agg({'Satin_Alma_Miktari': 'mean'})

Satin_Alma_Miktari
Kampanya                      
Kampanya A               15.02
Kampanya B               16.01

Matematiksel olarak interaktif ve kişiselleştirilmiş içerikler içeren (KAmpanya B) kampanyanin daha basarili oldugunu goruyoruz. Ancak bu sonucun istatistiksel olarak anlamlı olup olmadığını değerlendirmek için ileri istatistiksel analizler yapmamız gerekebilir.

SORU 4a - Doğru istatistiksel testin hangisi olduğuna karar verebilmek için sınamanız gereken varsayımlar nelerdir gösteriniz. Gerekli değilse varsayımı sınamayınız. (20 puan)

############################ #Normallik Varsayımı (shapiro) ############################

H0: Normal dağılım varsayımı sağlanmaktadır. 

H1: Normal dağılım varsayımı sağlanmamaktadır. 


Kriterler: 

Eğer p-değeri 0.01'den küçükse, null hipotezi (H0) reddedilir. 

Eğer p-değeri 0.01'den büyük veya eşitse, null hipotezi (H0) reddedilemez.

In [8]:
test_stat, pvalue = shapiro(df.loc[df['Kampanya'] == 'Kampanya A', 'Satin_Alma_Miktari'])
print("Test Stat = %.4f, p-value = %.4f" % (test_stat, pvalue))

Test Stat = 0.9766, p-value = 0.0716


In [9]:
test_stat, pvalue = shapiro(df.loc[df['Kampanya'] == 'Kampanya B', 'Satin_Alma_Miktari'])
print("Test Stat = %.4f, p-value = %.4f" % (test_stat, pvalue))

Test Stat = 0.9723, p-value = 0.0332


p-değerleri anlamlilik duzeyinden (0.01) büyük olduğu için, null hipotezi reddedilemez. Veriler normal dağılmaktadır. Diğer varsayımız olan Varyans Homojenliği varsayımına geçilir.

############################ Varyans Homojenliği Varsayımı (levene) ############################

H0: Grupların varyansları homojendir. 

H1: Grupların varyansları homojen değildir.


Kriterler: 

Eğer p-değeri 0.01'den küçükse, null hipotezi (H0) reddedilir. 

Eğer p-değeri 0.01'den büyük veya eşitse, null hipotezi (H0) reddedilemez.

In [10]:
test_stat, pvalue = levene(df.loc[df['Kampanya'] == 'Kampanya A', 'Satin_Alma_Miktari'],
                           df.loc[df['Kampanya'] == 'Kampanya B', 'Satin_Alma_Miktari'])
print("Test Stat = %.4f, p-value = %.4f" % (test_stat, pvalue))

Test Stat = 0.2637, p-value = 0.6081


Varyans homojenliği varsayımı da sağlanmıştır. Parametrik testi kullanmamız gerektiği sonucuna ulaşırız.

SORU 4b - Nihai test ve sonuçlarını gösteriniz. (15 puan)

Grup sayısı 2 olduğu için parametrik testlerden t testinin kullanılması gerekmektedir.

H0: Kampanya A ve Kampanya B'nin satın alma miktarları arasında istatistiksel olarak anlamlı bir fark yoktur.

H1: Kampanya A ve Kampanya B'nin satın alma miktarları arasında istatistiksel olarak anlamlı bir fark vardir.


ya da


H0: M0 — M1 = 0 

H1: M0 — M1 ≠ 0


Kriterler:

Eğer p-değeri 0.01'den küçükse, null hipotezi (H0) reddedilir. 

Eğer p-değeri 0.01'den büyük veya eşitse, null hipotezi (H0) reddedilemez

In [11]:
test_stat, pvalue =  ttest_ind(df.loc[df['Kampanya'] == 'Kampanya A', 'Satin_Alma_Miktari'],
                               df.loc[df['Kampanya'] == 'Kampanya B', 'Satin_Alma_Miktari'],
                               equal_var=True)
print("Test Stat = %.4f, p-value = %.4f" % (test_stat, pvalue))

Test Stat = -0.9273, p-value = 0.3549


H0 reddedilemez. Kampanya A ve Kampanya B'nin satın alma miktarları arasında istatistiksel olarak anlamlı bir fark yoktur.

SORU 4c	- Nihai test sonuçlarını dikkate alarak iş problemini yorumlayınız. (10 puan)

Kampanyalar farksizdir. Satin alma miktari uzerinde herhangi bir etkisi olmamaktadir.

SORU 5 - %5 anlamlılık düzeyi için 4. soruda elde ettiğiniz sonuçları tekrar yorumlayınız. Yalnızca gerekli olduğunu düşündüğünüz testleri tekrarlamanız beklenmektedir. Diğer türlü 4. soruda elde ettiğiniz test sonuçlarını kopyalayıp altına yorumlarınızı yazabilirsiniz. (30 puan)

############################ #Normallik Varsayımı (shapiro) ############################

H0: Normal dağılım varsayımı sağlanmaktadır. 

H1: Normal dağılım varsayımı sağlanmamaktadır. 


Kriterler: 

Eğer p-değeri 0.05'ten küçükse, null hipotezi (H0) reddedilir.

Eğer p-değeri 0.05'ten büyük veya eşitse, null hipotezi (H0) reddedilemez.


Kampanya B icin normallik testi p-value 0.0332'dir. Dolayisiyla yeni durumda (%5 anlamlilik duzeyi) normallik varsayiminin saglanamadigini goruyoruz. Parametrik olmayan testlerden iki grup ortalaması için kullanılan mannwhitneyu testi nihai test olarak kullanılacaktır.

############################ #Nihai Test (mannwhitneyu) ############################H0: Kampanya A ve Kampanya B'nin satın alma miktarları arasında istatistiksel olarak anlamlı bir fark yoktur.

H1: Kampanya A ve Kampanya B'nin satın alma miktarları arasında istatistiksel olarak anlamlı bir fark vardir.


ya da


H0: M0 — M1 = 0 

H1: M0 — M1 ≠ 0


Kriterler:

Eğer p-değeri 0.05'ten küçükse, null hipotezi (H0) reddedilir. 

Eğer p-değeri 0.05'ten büyük veya eşitse, null hipotezi (H0) reddedilemezdedilemez.

In [12]:
test_stat, pvalue =   mannwhitneyu(df.loc[df['Kampanya'] == 'Kampanya A', 'Satin_Alma_Miktari'],
                                   df.loc[df['Kampanya'] == 'Kampanya B', 'Satin_Alma_Miktari'])
print("Test Stat = %.4f, p-value = %.4f" % (test_stat, pvalue))

Test Stat = 4649.0000, p-value = 0.3912


H0 reddedilemez. Kampanya A ve Kampanya B'nin satın alma miktarları arasında istatistiksel olarak anlamlı bir fark yoktur. Kampanyalar farksizdir. 

BONUS: Seçtiğiniz iş probleminin nedenini kısaca açıklayınız. Kararınızı etkileyen faktörler nelerdir? (10 puan)